### ISSUE TITLE
Update variable group stopped working this afternoon. It looks like the task autoupgraded.

### ISSUE THREAD

**Issue by wyrdfish** [rfennell/AzurePipelines#530]
### Azure DevOps Extension you are using
https://github.com/rfennell/AzurePipelines/blob/master/Extensions/BuildUpdating/BuildVariableTask/BuildVariableTask.ps1

### Where are you running it?
- Azure DevOps Service (VSTS)
On private and hosted build agents.

### Version of Extension/Task
 Version: 1.15.1
(1.14.1 was working)
### Expected behaviour and actual behaviour

We have a step that updates a project specific build number in a library group. Has been working for months and stopped today around UK lunchtime.
Versions of the task have been updated and new ones are failing.
All our builds are broken.
now getting an error 
"Cannot set variable [ProjectBuildId] as variable could not be found in the a pipeline or associated variablegroups"

I rechecked config and tried completely new pipelines and variable groups.

### Steps to reproduce the problem
Set up a pipeline, add the task.
yaml (but we use old designer)
steps:
- task: richardfennellBM.BM-VSTS-BuildUpdating-Tasks.BuildVariableTask-Task.BuildVariableTask@1
  displayName: 'Update Build Variable'
  inputs:
    variable: ProjectBuildId
    value: '$(ProjectBuildId)'

Set up a variable group with correct permissions and a ProjectBuildId variable.

Run a build.

**Comment by fortunerayzor** [comment]
I want to confirm some bad behavior as well.
Version 1.15.1

In my case, all of my custom task conditions get wiped, bricking the build definition for further use.

**Comment by rfennell** [comment]
OK looks like the fix for #527 backwards compatibility has regressed the variable group update.

I will look at is ASAP, but I doubt it will be today

**Comment by rfennell** [comment]
This has been fixed in 1.16.1

I had introduced a regression, fixing the regression of losing support for TFS2017.

This has been fixed by adding logic to check what version of APIs are available at start of operations. This has been tested on TFS2017 and Azure DevOps for both variables and variable groups

@fortunerayzor  thanks for letting me know about this gap in my testing. I have revised my release process to include extra builds to test this use case.
